File: backend/src/main/java/com/hourglass/jobclocksync/model/Invoice.java
package com.hourglass.jobclocksync.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;
import java.util.List;

@Document(collection = "invoices")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Invoice {
    @Id
    private String id;
    
    @Indexed(unique = true)
    private String invoiceNumber;
    
    private String vendorId;
    private String companyId;
    private String workOrderId;
    private InvoiceStatus status;
    private Double totalAmount;
    private Double taxAmount;
    private Double subtotal;
    private List<InvoiceItem> items;
    private LocalDateTime dueDate;
    private LocalDateTime paidDate;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    
    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class InvoiceItem {
        private String description;
        private Integer quantity;
        private Double unitPrice;
        private Double total;
    }
    
    public enum InvoiceStatus {
        DRAFT, PENDING, APPROVED, PAID, REJECTED
    }
}

File: src/types/invoice.ts
export type InvoiceStatus = "DRAFT" | "PENDING" | "APPROVED" | "PAID" | "REJECTED";

export interface InvoiceItem {
  description: string;
  quantity: number;
  unitPrice: number;
  total: number;
}

export interface Invoice {
  id: string;
  invoiceNumber: string;
  vendorId: string;
  companyId: string;
  workOrderId: string;
  status: InvoiceStatus;
  totalAmount: number;
  taxAmount: number;
  subtotal: number;
  items: InvoiceItem[];
  dueDate?: string;
  paidDate?: string;
  createdAt: string;
  updatedAt: string;
}

export const INVOICE_STATUS_LABELS: Record<InvoiceStatus, string> = {
  DRAFT: "Draft",
  PENDING: "Pending",
  APPROVED: "Approved",
  PAID: "Paid",
  REJECTED: "Rejected",
};
